jwcrypto/common.py

```diff
@@ -104,3 +106,83 @@ class InvalidJWEKeyLength(JWException):
     def __init__(self, expected, obtained):
         msg = 'Expected key of lenght %d, got %d' % (expected, obtained)
         super(InvalidJWEKeyLength, self).__init__(msg)
+
+
+class InvalidJWSEHeaderRegistryOperation(JWException):
+    """Invalid JWSE Header Registry Operation.
+
+    This exception is raised when there is an error in trying ot add a JW
+    Signature or Encryption header to the Registry.
+    """
+
+    def __init__(self, message=None, exception=None):
+        msg = None
+        if message:
+            msg = message
+        else:
+            msg = 'Unknown Operation Failure'
+        if exception:
+            msg += ' {%s}' % repr(exception)
+        super(InvalidJWSEHeaderRegistryOperation, self).__init__(msg)
+
+
+# JWSE Header Registry definitions
+
+# RFC 7515 - 9.1: JSON Web Signature and Encryption Header Parameters Registry
+# HeaderParameters are for both JWS and JWE
+JWSEHeaderParameter = namedtuple('Parameter',
+                                'description mustprotect supported check_fn')
+
+class JWSEHeaderRegistry:
```

Comment:
shouldn't this simply extend 'dict' so that you do not ahve to redefine a dict-like interface, but only re-implement the __setitem__/__delitem__ calls ?

Reply:
I was hesitant about extending the `dict` class directly due to some dangers of weird built-in implementation specific issues. One of the few were:

http://www.kr41.net/2016/03-23-dont_inherit_python_builtin_dict_type.html

What do you think about this? I will defer that decision to you, I am okay with either but personal preference of not extending the `dict` class directly.

Reply:
Uhmm  ... interesting read.
Maybe use the Mutable thing instead ?
We already import namedtuple from collections, so that is ok to have as a dependency I guess.

Reply:
Ok sg, i'll use that then.